otherlibs/dune-rpc/dune_rpc.mli

```diff
@@ -135,8 +170,9 @@ module V1 : sig
 
       val create :
            ?log:(Message.t -> unit fiber)
-        -> ?errors:(Error.t list -> unit fiber)
-        -> ?promotions:(Promotion.t list -> unit fiber)
+        -> ?diagnostic:(Diagnostic.Event.t list -> unit fiber)
```

Comment:
Should we add some documentation guaranteeing that the first call to this callback will include every currently-existing error being `Add`ed? If we're also removing the `diagnostics` request, it might be a bit confusing for clients who would like to connect once, get the list of errors, then immediately disconnect.

Reply:
> Should we add some documentation guaranteeing that the first call to this callback will include every currently-existing error being Added?

Will do.

> If we're also removing the diagnostics request, it might be a bit confusing for clients who would like to connect once, get the list of errors, then immediately disconnect.

If there's demand for clients to do this, I will add it back.